components/org.wso2.carbon.identity.application.authenticator.google/src/main/java/org/wso2/carbon/identity/application/authenticator/google/Utils.java

```diff
@@ -77,6 +89,16 @@ public static boolean validateGoogleJWT(String idToken, String audience)
 
         // Verifying the issuers and audiences.
         if (claimsSet != null && claimsSet.toJSONObject() != null && !claimsSet.toJSONObject().isEmpty()) {
+
+            if (internalSubmission){
+                String nonceFromGoogle = String.valueOf(claimsSet.getClaim(NONCE));
+                if (!nonce.equals(nonceFromGoogle)){
```

Comment:
check whether `nonce` is non-null.

Reply:
Fixed 414aed35225d4d7f2cf7ef2d89e7daddc3ca3a68